pootle/static/js/editor/app.js

```diff
@@ -2315,4 +2373,55 @@ PTL.editor = {
     return true;
   },
 
+  toggleSuggestion(e, { canHide = false } = {}) {
+    e.stopPropagation();
+    const suggestionId = parseInt(e.currentTarget.dataset.suggId, 10);
+    const suggestion = document.getElementById(`suggestion-${suggestionId}`);
+
+    if (this.selectedSuggestionId === undefined) {
+      this.selectedSuggestionId = suggestionId;
+      const props = {
+        suggId: this.selectedSuggestionId,
+        languageDirection: this.settings.languageDirection,
+        initialSuggestionText: e.currentTarget.dataset.translationAid,
+        onAcceptSuggestion: this.handleAcceptSuggestion,
+        onRejectSuggestion: this.handleRejectSuggestion,
+        onChange: this.handleSuggestionFeedbackChange,
+
+      };
+      const feedbackMountPoint = document.querySelector('.js-mnt-suggestion-feedback');
+      const editorBody = document.querySelector('.js-editor-body .translate-full');
+      suggestion.classList.add('suggestion-expanded');
+      editorBody.classList.add('suggestion-expanded');
+
+      this.isSuggestionFeedbackFormDirty = false;
+      this.suggestionFeedbackForm = ReactDOM.render(
+        <SuggestionFeedbackForm {...props} />,
+        feedbackMountPoint
+      );
+    } else if (canHide) {
+      this.closeSuggestion();
+    }
+  },
+
+  handleSuggestionFeedbackChange(isDirty) {
+    PTL.editor.isSuggestionFeedbackFormDirty = isDirty;
+  },
+
+  closeSuggestion({ checkIfCanNavigate = true } = {}) {
+    if (this.selectedSuggestionId !== undefined &&
+        (!checkIfCanNavigate || this.canNavigate())) {
+      const suggestion = document.querySelector(`#suggestion-${this.selectedSuggestionId}`);
+      const editorBody = document.querySelector('.js-editor-body .translate-full');
+      const feedbackMountPoint = document.querySelector('.js-mnt-suggestion-feedback');
+      const node = feedbackMountPoint.childNodes[0];
+      editorBody.classList.remove('suggestion-expanded');
+      suggestion.classList.remove('suggestion-expanded');
+      ReactDOM.unmountComponentAtNode(node);
```

Comment:
Note that `node` is unnecessary (in fact, it doesn't work), and `unmountComponentAtNode()` needs to receive the original node where the component was mounted, `feedbackMountPoint`. The rest is handled by React, and there's no need to deal with the DOM via `removeChild()` or similar 😃.